### ISSUE TITLE
How to PATCH a related entity?

### ISSUE THREAD

**Issue by mcanepa** [api-platform/api-platform#2620]
I do a `GET` to `/contacts/5` and I get

```
{
    "@context": "/contexts/Contact",
    "@id": "/contacts/5",
    "@type": "Contact",
    "id": 5,
    "firstName": "MyFirstName",
    "lastName": "MyLastName",
    "emails": [
        {
            "@type": "Email",
            "@id": "/.well-known/genid/7f6e92c949360d568205",
            "id": 10,
            "address": "mail10@example.com",
            "type": "personal"
        },
        {
            "@type": "Email",
            "@id": "/.well-known/genid/ca7e900a8365cb57b24f",
            "id": 20,
            "address": "mail20@example.com",
            "type": "personal"
        }
    ]
}
```

then I want to `PATCH` the second email and tried to do

```
{
    "emails": [
        {
            "id": 20,
            "address": "mail20_edited@example.com"
        }
    ]
}
```

or even

```
{
    "emails": [
        {
            "@id": "/emails/20",
            "address": "mail20_edited@example.com"
        }
    ]
}
```

but it deletes all the emails (for contact 5) and create a new one.

What is the correct way to do it?

Is it possible? I've read a couple of issues where they are struggling with this (#2096 and [#5559](https://github.com/api-platform/core/issues/5559))

**Comment by mcanepa** [comment]
I've also tried alleged fix [#5577](https://github.com/api-platform/core/pull/5577) but it doesn't work either

**Comment by mcanepa** [comment]
After much trial and error I've got it to work by issuing a `PUT` request with this content

```
{
    "@id": "/contacts/5",
    "emails": [
        {
            "@id": "/emails/20",
            "address": "mail20_edited@example.com"
        }
    ]
}
```

That is an IRI for the contact and an IRI for the email!

Note that any other emails will be deleted. If you want to keep them, then do this

```
{
    "@id": "/contacts/5",
    "emails": [
        "/emails/10",
        {
            "@id": "/emails/20",
            "address": "mail20_edited@example.com"
        }
    ]
}
```
